exonum/src/encoding/protobuf/mod.rs

```diff
@@ -247,7 +218,29 @@ impl ProtobufConvert for Vec<u8> {
     fn to_pb(&self) -> Self::ProtoStruct {
         self.clone()
     }
-    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, ()> {
+    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error> {
         Ok(pb)
     }
 }
+
+macro_rules! impl_protobuf_convert_scalar {
+    ($name:tt) => {
+        impl ProtobufConvert for $name {
+            type ProtoStruct = $name;
+            fn to_pb(&self) -> Self::ProtoStruct {
+                *self
+            }
+            fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error> {
+                Ok(pb)
+            }
+        }
+    };
+}
+
+impl_protobuf_convert_scalar!(bool);
+impl_protobuf_convert_scalar!(u32);
+impl_protobuf_convert_scalar!(u64);
+impl_protobuf_convert_scalar!(i32);
+impl_protobuf_convert_scalar!(i64);
```

Comment:
What about `u8`, `u16`, `i8`, `i16`?

Reply:
u8 will not be supported since Vec<u8> is specific case for protobuf bytes.(conflict with Vec<T: ProtobufConvert>)
other one should be converted with bound checks.